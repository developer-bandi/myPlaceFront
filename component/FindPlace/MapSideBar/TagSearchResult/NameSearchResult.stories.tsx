import { ComponentStory, ComponentMeta } from "@storybook/react";
import TagSearchResult from "./TagSearchResult";
import { action } from "@storybook/addon-actions";

export default {
  title: "FindPlace/MapSideBar/TagSearchResult",
  component: TagSearchResult,
  parameters: {
    layout: "fullscreen",
  },
} as ComponentMeta<typeof TagSearchResult>;

const Template: ComponentStory<typeof TagSearchResult> = (arg) => {
  return (
    <div style={{ width: 340, position: "relative" }}>
      <TagSearchResult
        {...arg}
        searchCondition={{
          position: {},
          category: "카페",
          hashtag: ["맛있는"],
        }}
      />
    </div>
  );
};

export const isLoading = Template.bind({});
isLoading.args = {
  searchResult: { loading: true, error: false },
};

export const isError = Template.bind({});
isError.args = {
  searchResult: { loading: false, error: true },
};

export const isBeforeSearch = Template.bind({});
isBeforeSearch.args = {
  searchResult: { loading: false, error: false },
};

export const isNoResult = Template.bind({});
isNoResult.args = {
  searchResult: { content: [], loading: false, error: false },
};

export const isNormal = Template.bind({});
isNormal.args = {
  searchResult: {
    content: [
      {
        id: 1,
        name: "상점",
        category: "카페",
        latitude: "test",
        longitude: "test",
        dist: 10,
        hashtag: {
          아름다운: 5,
          깨끗한: 4,
        },
      },
    ],
    loading: false,
    error: false,
  },
};
