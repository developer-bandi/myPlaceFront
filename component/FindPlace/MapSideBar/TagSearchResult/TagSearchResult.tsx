import styles from "./TagSearchResult.module.scss";
import searchResultLoading from "../../../../public/searchResultLoading.gif";
import StoreBox from "../StoreBox/StoreBox";
import { hashtagSearchConditionState } from "../../../../store/reducers/searchCondition/Reducer";
import { SearchResultState } from "../../../../store/reducers/searchResult/Reducer";
import Image from "next/image";
import TopInfo from "./topInfo/TopInfo";

interface TagSearchResultProps {
  moveSearhPage: () => void;
  searchResult: SearchResultState;
  searchCondition: hashtagSearchConditionState;
  showStoreInfo: (storeId: number) => void;
}

const TagSearchResult = ({
  moveSearhPage,
  searchResult,
  searchCondition,
  showStoreInfo,
}: TagSearchResultProps) => {
  return (
    <>
      <TopInfo
        moveSearhPage={moveSearhPage}
        searchCondition={searchCondition}
      />
      <div className={styles.storeListBlock}>
        {searchResult.loading ? (
          <div className={styles.loading} data-testid="loading">
            <Image src={searchResultLoading} alt="loading"></Image>
          </div>
        ) : searchResult.error ? (
          <p className={styles.ectStatus} data-testid="error">
            에러가 발생하였습니다
          </p>
        ) : searchResult.content?.length === 0 ? (
          <p className={styles.ectStatus} data-testid="noResult">
            검색결과가 없습니다
          </p>
        ) : (
          <StoreBox
            searchResult={searchResult.content}
            showStoreInfo={showStoreInfo}
          />
        )}
      </div>
    </>
  );
};

export default TagSearchResult;
