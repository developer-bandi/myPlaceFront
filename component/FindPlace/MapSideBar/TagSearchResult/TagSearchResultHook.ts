import { useDispatch, useSelector } from "react-redux";
import { RootReducer } from "../../../../store";
import { setSearchType } from "../../../../store/reducers/searhType/Reducer";
import { getStoreInfo } from "../../../../store/reducers/storeInfo/Reducer";

const useTagSearchResult = () => {
  const searchResult = useSelector((state: RootReducer) => state.searchResult);
  const searchCondition = useSelector(
    (state: RootReducer) => state.searchCondition
  );
  const dispatch = useDispatch();
  const showStoreInfo = (storeId: number) => {
    dispatch(getStoreInfo(storeId));
  };

  const moveSearhPage = () => {
    dispatch(setSearchType("hashtagSearch"));
  };

  return {
    moveSearhPage,
    searchResult,
    searchCondition,
    showStoreInfo,
  };
};

export default useTagSearchResult;
