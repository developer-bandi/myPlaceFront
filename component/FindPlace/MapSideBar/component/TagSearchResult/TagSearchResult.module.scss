.topBlock {
  border-bottom: 1px solid #e5e5e5;
  padding: 20px 0 10px 0;
}

.titleBlock {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 0 20px 20px 20px;
}

.movesearch {
  height: 30px;
  border-radius: 10px;
  color: white;
  padding: 5px;
  background-color: skyblue;
}

.title {
  color: #00aaff;
}

.selecttaglistBlock {
  display: flex;
  flex-wrap: wrap;
  padding-left: 20px;
  color: gray;
  overflow-x: scroll;
}

.selecttag {
  margin-right: 10px;
  margin-bottom: 10px;
}

.ectStatus {
  text-align: center;
  font-size: 25px;
  margin-top: 40px;
  color: gray;
}

.loading {
  margin: 200px 100px 0 100px;
}

.storeListBlock {
  height: 750px;
  overflow-y: scroll;
  @media screen and (max-width: 767px) {
    height: 350px;
  }
}
