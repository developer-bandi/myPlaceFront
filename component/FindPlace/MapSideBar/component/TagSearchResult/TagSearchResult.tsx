import styles from "./TagSearchResult.module.scss";
import searchResultLoading from "../../../../../public/searchResultLoading.gif";
import StoreBox from "../Common/NameSearchResult/StoreBox";
import {hashtagSearchConditionState} from "../../../../../store/reducers/hashtagSearchCondition/Reducer";
import {SearchResultState} from "../../../../../store/reducers/searchResult/Reducer";
import Image from "next/image";

interface TagSearchResultProps {
  moveSearhPage: () => void;
  searchResult: SearchResultState;
  searchCondition: hashtagSearchConditionState;
  showStoreInfo: (storeId: number) => void;
}

const TagSearchResult = ({
  moveSearhPage,
  searchResult,
  searchCondition,
  showStoreInfo,
}: TagSearchResultProps) => {
  return (
    <>
      <div className={styles.topBlock}>
        <div className={styles.titleBlock}>
          <h2 className={styles.title}>검색결과</h2>
          <button
            className={styles.movesearch}
            onClick={moveSearhPage}
            data-testid="moveSearhPage"
          >
            태그 재선택하기
          </button>
        </div>
        <ul className={styles.selecttaglistBlock}>
          {[searchCondition.category, ...searchCondition.hashtag].map(
            (data: string, index) => {
              return (
                <li className={styles.selecttag} key={index}>
                  #{data}
                </li>
              );
            }
          )}
        </ul>
      </div>
      <div className={styles.storeListBlock}>
        {searchResult.loading ? (
          <div className={styles.loading} data-testid="loading">
            <Image src={searchResultLoading} alt="loading"></Image>
          </div>
        ) : searchResult.error ? (
          <p className={styles.ectStatus} data-testid="error">
            에러가 발생하였습니다
          </p>
        ) : searchResult.content === undefined ? (
          <p className={styles.ectStatus} data-testid="noResult">
            검색결과가 없습니다
          </p>
        ) : (
          <StoreBox
            searchResult={searchResult.content}
            showStoreInfo={showStoreInfo}
          />
        )}
      </div>
    </>
  );
};

export default TagSearchResult;
