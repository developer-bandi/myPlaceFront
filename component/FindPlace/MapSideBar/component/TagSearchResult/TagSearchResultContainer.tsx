import {useDispatch, useSelector} from "react-redux";
import {RootReducer} from "../../../../../store";
import {setSearchType} from "../../../../../store/reducers/searhType/Reducer";
import {getStoreInfo} from "../../../../../store/reducers/storeInfo/Reducer";
import TagSearchResult from "./TagSearchResult";

const TagSearchResultContainer = () => {
  const searchResult = useSelector((state: RootReducer) => state.searchResult);
  const searchCondition = useSelector(
    (state: RootReducer) => state.hashtagSearchCondition
  );
  const dispatch = useDispatch();
  const showStoreInfo = (storeId: number) => {
    dispatch(getStoreInfo(storeId));
  };

  const moveSearhPage = () => {
    dispatch(setSearchType({type: "hashtag", hashtag: "search"}));
  };

  return (
    <TagSearchResult
      moveSearhPage={moveSearhPage}
      searchResult={searchResult}
      searchCondition={searchCondition}
      showStoreInfo={showStoreInfo}
    ></TagSearchResult>
  );
};

export default TagSearchResultContainer;
