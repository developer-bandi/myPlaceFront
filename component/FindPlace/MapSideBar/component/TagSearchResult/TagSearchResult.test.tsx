import {
  act,
  fireEvent,
  render,
  renderHook,
  screen,
} from "@testing-library/react";
import {Provider} from "react-redux";
import configureMockStore from "redux-mock-store";
import newReact from "react";
import * as React from "react";
import useTagSearchResult from "./TagSearchResultHook";
import TagSearchResult from "./TagSearchResult";

const mockStore = configureMockStore()({
  searchResult: {},
  hashtagSearchCondition: {},
});
const wrapper = ({children}: any) => (
  <Provider store={mockStore}>{children}</Provider>
);

describe("TagSearchResult Hook 테스트", () => {
  it("showStoreInfo 함수 테스트", () => {
    const {result} = renderHook(() => useTagSearchResult(), {
      wrapper,
    });
    act(() => {
      result.current.showStoreInfo(1);
    });
    expect(mockStore.getActions()[0]).toEqual({
      payload: 1,
      type: "storeInfo/getStoreInfo",
    });
  });
  it("moveSearhPage 함수 테스트", () => {
    const {result} = renderHook(() => useTagSearchResult(), {
      wrapper,
    });
    act(() => {
      result.current.moveSearhPage();
    });
    expect(mockStore.getActions()[1]).toEqual({
      payload: "hashtagSearch",
      type: "searchType/setSearchType",
    });
  });
});

describe("TagSearchResult Presentational 테스트", () => {
  it("로딩중", () => {
    const moveSearhPageMock = jest.fn();
    const showStoreInfoMock = jest.fn();
    const utils = render(
      <TagSearchResult
        moveSearhPage={moveSearhPageMock}
        searchResult={{loading: true, error: false}}
        searchCondition={{
          category: "testCategory",
          position: {
            address: "testAddress",
            longitude: "testLongitude",
            latitude: "testLatitude",
          },
          hashtag: ["testHashtag"],
          keyword: "testKeyword",
        }}
        showStoreInfo={showStoreInfoMock}
      ></TagSearchResult>
    );
    expect(utils.container).toMatchSnapshot();
    screen.getByTestId("loading");
  });
  it("에러발생", () => {
    const moveSearhPageMock = jest.fn();
    const showStoreInfoMock = jest.fn();
    const utils = render(
      <TagSearchResult
        moveSearhPage={moveSearhPageMock}
        searchResult={{loading: false, error: true}}
        searchCondition={{
          category: "testCategory",
          position: {
            address: "testAddress",
            longitude: "testLongitude",
            latitude: "testLatitude",
          },
          hashtag: ["testHashtag"],
          keyword: "testKeyword",
        }}
        showStoreInfo={showStoreInfoMock}
      ></TagSearchResult>
    );
    expect(utils.container).toMatchSnapshot();
    screen.getByTestId("error");
  });
  it("컨텐츠 없음", () => {
    const moveSearhPageMock = jest.fn();
    const showStoreInfoMock = jest.fn();
    const utils = render(
      <TagSearchResult
        moveSearhPage={moveSearhPageMock}
        searchResult={{content: [], loading: false, error: false}}
        searchCondition={{
          category: "testCategory",
          position: {
            address: "testAddress",
            longitude: "testLongitude",
            latitude: "testLatitude",
          },
          hashtag: ["testHashtag"],
          keyword: "testKeyword",
        }}
        showStoreInfo={showStoreInfoMock}
      ></TagSearchResult>
    );
    expect(utils.container).toMatchSnapshot();
    screen.getByTestId("noResult");
  });
  it("컨텐츠 있음", () => {
    const moveSearhPageMock = jest.fn();
    const showStoreInfoMock = jest.fn();
    const utils = render(
      <TagSearchResult
        moveSearhPage={moveSearhPageMock}
        searchResult={{
          content: [
            {
              id: 1,
              name: "testName",
              category: "testCategory",
              latitude: "testLatitude",
              longitude: "testLongitude",
              dist: 1,
              hashtag: {
                testHashtag: 1,
              },
            },
          ],
          loading: false,
          error: true,
        }}
        searchCondition={{
          category: "testCategory",
          position: {
            address: "testAddress",
            longitude: "testLongitude",
            latitude: "testLatitude",
          },
          hashtag: ["testHashtag"],
          keyword: "testKeyword",
        }}
        showStoreInfo={showStoreInfoMock}
      ></TagSearchResult>
    );
    expect(utils.container).toMatchSnapshot();
    fireEvent.click(screen.getByTestId("moveSearhPage"));
    expect(moveSearhPageMock).toBeCalledTimes(1);
  });
});
