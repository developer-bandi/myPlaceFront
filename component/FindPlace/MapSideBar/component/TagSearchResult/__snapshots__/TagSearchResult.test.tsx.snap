// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagSearchResult Presentational 테스트 로딩중 1`] = `
<div>
  <div
    class="topBlock"
  >
    <div
      class="titleBlock"
    >
      <h2
        class="title"
      >
        검색결과
      </h2>
      <button
        class="movesearch"
        data-testid="moveSearhPage"
      >
        태그 재선택하기
      </button>
    </div>
    <ul
      class="selecttaglistBlock"
    >
      <li
        class="selecttag"
      >
        #
        testCategory
      </li>
      <li
        class="selecttag"
      >
        #
        testHashtag
      </li>
    </ul>
  </div>
  <div
    class="storeListBlock"
  >
    <div
      class="loading"
      data-testid="loading"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
      >
        <span
          style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
            style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
          />
        </span>
        <img
          alt="loading"
          data-nimg="intrinsic"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
    </div>
  </div>
</div>
`;

exports[`TagSearchResult Presentational 테스트 에러발생 1`] = `
<div>
  <div
    class="topBlock"
  >
    <div
      class="titleBlock"
    >
      <h2
        class="title"
      >
        검색결과
      </h2>
      <button
        class="movesearch"
        data-testid="moveSearhPage"
      >
        태그 재선택하기
      </button>
    </div>
    <ul
      class="selecttaglistBlock"
    >
      <li
        class="selecttag"
      >
        #
        testCategory
      </li>
      <li
        class="selecttag"
      >
        #
        testHashtag
      </li>
    </ul>
  </div>
  <div
    class="storeListBlock"
  >
    <p
      class="ectStatus"
      data-testid="error"
    >
      에러가 발생하였습니다
    </p>
  </div>
</div>
`;

exports[`TagSearchResult Presentational 테스트 컨텐츠 없음 1`] = `
<div>
  <div
    class="topBlock"
  >
    <div
      class="titleBlock"
    >
      <h2
        class="title"
      >
        검색결과
      </h2>
      <button
        class="movesearch"
        data-testid="moveSearhPage"
      >
        태그 재선택하기
      </button>
    </div>
    <ul
      class="selecttaglistBlock"
    >
      <li
        class="selecttag"
      >
        #
        testCategory
      </li>
      <li
        class="selecttag"
      >
        #
        testHashtag
      </li>
    </ul>
  </div>
  <div
    class="storeListBlock"
  >
    <p
      class="ectStatus"
      data-testid="noResult"
    >
      검색결과가 없습니다
    </p>
  </div>
</div>
`;

exports[`TagSearchResult Presentational 테스트 컨텐츠 있음 1`] = `
<div>
  <div
    class="topBlock"
  >
    <div
      class="titleBlock"
    >
      <h2
        class="title"
      >
        검색결과
      </h2>
      <button
        class="movesearch"
        data-testid="moveSearhPage"
      >
        태그 재선택하기
      </button>
    </div>
    <ul
      class="selecttaglistBlock"
    >
      <li
        class="selecttag"
      >
        #
        testCategory
      </li>
      <li
        class="selecttag"
      >
        #
        testHashtag
      </li>
    </ul>
  </div>
  <div
    class="storeListBlock"
  >
    <p
      class="ectStatus"
      data-testid="error"
    >
      에러가 발생하였습니다
    </p>
  </div>
</div>
`;
